src/ResourceManager/LogicApp/Commands.LogicApp/Cmdlets/IntegrationAccount/GetAzureIntegrationAccountGeneratedX12IcnCommand.cs

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.LogicApp.Cmdlets
+{
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.LogicApp.Utilities;
+
+    /// <summary>
+    /// Gets the integration account generated X12 interchange control number by agreement name
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmIntegrationAccountGeneratedX12Icn")]
+    [OutputType(typeof(IntegrationAccountControlNumber), typeof(IList<QualifiedIntegrationAccountControlNumber>))]
+    public class GetAzureIntegrationAccountGeneratedX12IcnCommand : LogicAppBaseCmdlet
+    {
+        #region Input Parameters
+
+        /// <summary>
+        /// Gets or sets the resource group name.
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "The integration account resource group name.",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the integration account name.
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "The integration account name.")]
+        [Alias("ResourceName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agreement name.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "The integration account agreement name.")]
+        [ValidateNotNullOrEmpty]
+        public string AgreementName { get; set; }
+
+        #endregion Input Parameters
+
+        /// <summary>
+        /// Executes the get integration account X12 control number command.
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            this.WriteObject(
+                sendToPipeline: string.IsNullOrEmpty(this.AgreementName) ?
+                    this.IntegrationAccountClient.ListIntegrationAccountGeneratedIcns(
+                        resourceGroupName: this.ResourceGroupName,
+                        integrationAccountName: this.Name,
+                        agreementType: Management.Logic.Models.AgreementType.X12) as object :
```

Comment:
@cormacpayne 
The reason is we don't want to break the existent behavior: if no switch parameter is provided. 
Since the current implementation only returns X12 control numbers, if we take X12 as the default parameter (we feel it will confuse the user mistakenly updating an Edifact control number instead of X12)
However, if we force the switch parameter as required, it would be a breaking change (need 6 months prior notice as by powershell rule).
Therefore we decided to add 2 new cmdlets for X12/Edifact and will start the deprecation process for the existent one.